### DISCUSSION TITLE
About the status of atlas support for regional domains

### DISCUSSION THREAD

**Discussion by TingLei-daprediction** [ecmwf/atlas#190] [General]
Dear Atlas experts,
What is the status on the atlas support for regional domains?  any more complete and updated doc?  While, I am going to dig and learn from examples as atlas used in jedi,  That will be very helpful if any explanation/instruction given here. 
For example, one of my current questions is a class regularregionalgrid is mentioned in https://sites.ecmwf.int/docs/atlas/design/grid/#regularregionalgrid.   But it seems not available in the current atlas version. 
Your comments/suggestions on this is appreciated.

**Comment by MarekWlasak** [comment]
See 

https://github.com/ecmwf/atlas/blob/55e20456d5f754e9d6e1180d8db5fa866010bf3b/doc/example-grids/regional_lonlat_1.yml#L4

Looks like its name may have changed.

**Comment by MarekWlasak** [comment]
Note that the best way to learn atlas is to look through and understand the ctests.

**Comment by TingLei-daprediction** [comment]
@MarekWlasak  So many thanks!  I will follow those regional examples you point to see how it work for regional grids.

**Comment by TingLei-daprediction** [comment]
Per @benjaminmenetrier permission, his answer relevant to this topic  is quoted below:
```
The ATLAS interpolator is not implemented if the source is a regional StructuredGrid. It crashes in the [MatchingMeshPartitioner](https://github.com/ecmwf/atlas/blob/55e20456d5f754e9d6e1180d8db5fa866010bf3b/src/atlas/grid/detail/partitioner/MatchingMeshPartitionerLonLatPolygon.cc#L56).
I've implemented my own interpolator for regional grids [here](https://git.ecmwf.int/projects/SAB/repos/quench/browse/src/quench/interpolation).
You should use an ATLAS version >= 0.37.0, where https://github.com/ecmwf/atlas/pull/172 is merged.
```
